### ISSUE TITLE
Grand Theft Auto V massive performance drop if on desert

### ISSUE THREAD

**Issue by kassindornelles** [doitsujin/dxvk#3311]
If you are in the desert and you look at the direction of the city the game performance completely falls apart, see the images:

i'm in the desert looking at the city direction:
![hudmap](https://user-images.githubusercontent.com/40970965/227653200-460f96c0-7a15-4a1c-9a9e-2287c21eabe9.png)

![gta](https://user-images.githubusercontent.com/40970965/227653260-c942d234-fe7b-488a-b715-bf4a14e3c20b.png)

### Software information
Grand Theft Auto V - (Steam)

### System information
- GPU: AMD RX 6600
- Driver: Mesa 23.0.0 (RADV)
- Wine version: Proton Experimental (24/04/2023)
- DXVK version: 2.1

### Apitrace file(s)
- Unfortunately looks like the game DRM does not like people apitracing it, i tried many times before but since i'm stupid you guys might have more luck than me to capture it

### Log files
[steam-271590.tar.gz](https://github.com/doitsujin/dxvk/files/11066902/steam-271590.tar.gz)

**Comment by Trevonn** [comment]
Reproduced on my 6600M
Very high Settings

### System information
- GPU: RX 6600M
- CPU: R5 5600H
- Driver: Mesa 23.0.0 (RADV)
- Wine version: 8.4 (Staging)
- DXVK version: 2.1

**Comment by doitsujin** [comment]
Is there actually any indication that dxvk is at fault here? `perf top` might show something.

**Comment by Trevonn** [comment]
![Screenshot_20230325_131357](https://user-images.githubusercontent.com/3117355/227719300-f2f32e48-5cb7-4753-8782-dc5189e36a62.png)
`d3d11.cachedDynamicResources = a` noticeably improves performance for me

**Comment by doitsujin** [comment]
looks like we need an app profile then, can you try narrowing down which option exactly it is that improves performance?

Please try each of:
- `d3d11.cachedDynamicResources = c`
- `d3d11.cachedDynamicResources = r`
- `d3d11.cachedDynamicResources = vi`

If any of those give you full performance, we should use that.

**Comment by kassindornelles** [comment]
Both 
`d3d11.cachedDynamicResources = vi` and  `d3d11.cachedDynamicResources = a`
boosted my performance to normal levels (max settings 60~70/80 fps)
I tested both separately of course.

Framerate in the screenshot is on the bottom left*
![A](https://user-images.githubusercontent.com/40970965/227743147-56a9aabd-7708-4930-aa42-7635f3f7bfd1.png)

**Comment by adolfintel** [comment]
Oh wow, that improved performance way more than I expected (6900xt)

**Comment by falcon35180** [comment]
I have a few screenshots showing a huge performance difference, even with all cached dynamic resources options set. Not sure if it's related to DXVK, the OS or the driver, but hopefully someone has some ideas about where to look.

I wish I could use DXVK with the game on Windows to rule out some variables, but it doesn't seem to work - as I remember, it always loads dxgi.dll from System32.

Also, if anyone with an AMD GPU could test in the same location in the game, that could give us another useful data point. I can provide map and graphics settings screenshots if necessary.

Anyway, the hardware is:

- Ryzen 9 5900x
- 2x32GB DDR4-3200
- MSI RTX 3080 Ti 3X Ventus OC

Windows 10:
![20230510125748_1](https://github.com/doitsujin/dxvk/assets/89119559/69233267-aff3-408d-9b5c-7a16bbc10ff6)

Linux Mint 21.1 Cinnamon:
![20230510130909_1](https://github.com/doitsujin/dxvk/assets/89119559/c3fcea69-9a19-4c9f-8374-d84872cacc6c)
![20230510130912_1](https://github.com/doitsujin/dxvk/assets/89119559/acc629d3-2a17-41a3-bbf6-e332aed08e2a)
![20230510130924_1](https://github.com/doitsujin/dxvk/assets/89119559/77753954-ada6-42bf-aee3-b6a3c9c6b002)
(I included this last one that hits the FPS cap for comparison)

**Comment by Blisto91** [comment]
I don't know where that is and haven't really played GTA V much. Can you provide a savegame near the location or do i have to get there?

**Comment by falcon35180** [comment]
I don't know if there are any complications with sharing savegames, given that there are Rockstar and Steam accounts involved. I'll try to find out more.

In the meantime, I'll post a map screenshot when I get a chance.

**Comment by mbriar** [comment]
> I have a few screenshots showing a huge performance difference, even with all cached dynamic resources options set

**Comment by falcon35180** [comment]
> Does setting `d3d11.cachedDynamicResources = a` fix this?

I tried "vi" and "a" and got the same results. It did look promising at first, during the benchmark sequence and in some other situations, but then I came across this.

I seem to remember trying it with the NVAPI hack disabled in the past, but it didn't help. I might check that again just to confirm.

I have tested it on another PC with a Vega 56 GPU to try to rule out any Nvidia driver problems - performance was on par with Windows when GPU bound but turning down graphics settings to load the CPU more showed similar disparities. That PC has a Core i7-6700 CPU - at one point, that was my main PC with the 3080 Ti (yes, that GPU was kind of an overkill, but I did use it for crypto mining as well!)

As I mentioned earlier, it's a pity I can't get the game to run with DXVK libraries on Windows, since I'd love to see it without OS and driver differences getting in the way.


> Honestly I'm tired of dealing with this game's crap all the time

I know the feeling! At least it's good enough at the moment to avoid rebooting if I feel like playing. And it's nowhere near as bad as Halo Infinite!
